apps/api/src/routers/admin.py

```diff
@@ -125,12 +125,68 @@ async def release_decisions() -> None:
         group = [record for record in records if record["decision"] == decision]
         if not group:
             continue
-
         await asyncio.gather(
             *(_process_batch(batch, decision) for batch in batched(group, 100))
         )
 
 
+@router.post(
+    "/confirm-attendance", dependencies=[Depends(require_role([Role.DIRECTOR]))]
+)
+async def confirm_attendance() -> None:
+    """Update applicant status to void or attending based on their current status."""
+    records = await mongodb_handler.retrieve(
+        Collection.USERS,
+        {"role": Role.APPLICANT},
+        ["_id", "status"],
+    )
+
+    confirmed_record = [
+        record for record in records if record["status"] == Status.CONFIRMED
+    ]
+    for record in confirmed_record:
+        record["status"] = Status.ATTENDING
+
+    await asyncio.gather(
+        *(
+            _process_status(batch, Status.ATTENDING)
+            for batch in batched(confirmed_record, 100)
+        )
+    )
+
+    for initial_status in (Status.WAIVER_SIGNED, Decision.ACCEPTED):
+        void_record = [
+            record for record in records if record["status"] == initial_status
+        ]
+
+        for record in void_record:
+            record["status"] = Status.VOID
+        await asyncio.gather(
+            *(
+                _process_status(batch, Status.VOID)
+                for batch in batched(void_record, 100)
+            )
+        )
```

Comment:
Suggestion: instead of repeating this logic, we could use a mapping of `status` to `new_status` to describe the enforcement process.
```python
{
    Status.CONFIRMED: Status.ATTENDING,
    Status.ACCEPTED: Status.VOID,
    Status.WAIVER_SIGNED: Status.VOID,
}